### PR TITLE
Update wxMemoryInputStream::Peek to set last read count.

### PR DESCRIPTION
**Pull request by the-nerbs**
Fix for #23757 

This updates the implementation of `wxMemoryInputStream::Peek` to set the last read count to 1 if it successfully retrieved a byte from the stream, or 0 if it did not. This change is to align the memory stream's behavior with that of the other stream types - `wxFileInputStream`, `wxFFileInputStream`, or `wxStringInputStream` for a few examples.

### CODE REVIEW

File: tests/streams/bstream.h
Comment (the-nerbs) [inline tests/streams/bstream.h:256]: I want to highlight this test update as something I'm not entirely sure about, since I believe it would impose this behavior of `Peek` on all stream types. This seems to be true at the moment, though as PB mentioned on the issue, this behavior is not formally documented.

It is mentioned by the in-source comments on `wxInputStream::LastRead` that `Peek` affects it, and on `wxInputStream::Peek` that the last read count is set to 0 or 1 though, which is why I thought to update the common tests here instead of just adding one against the memory stream specifically.

I'm wondering if I should change this test to just be against the memory stream, or perhaps this change might warrant an update to the doxygen comments for `wxInputStream::Peek` and/or `LastRead` as well?

Comment (vadz) [reply tests/streams/bstream.h:256]: It does make sense to impose the same behaviour for all stream types, however, and as all the existing tests pass, it seems that this is already the case, so I think it's fine to do this.

**Review by vadz — APPROVED** [review]
Thanks for fixing this, I agree that it's a change for the better and so will merge it in master soon if there are no objections. I probably won't merge it into 3.2 however just in case somebody relies on the old behaviour there (even though I have no idea why would anybody do it).

I'll also update `Peek()` documentation by basically copying the source comment contents there.

**Comment by PBfordev** [comment]
> I'll also update `Peek()` documentation by basically copying the source comment contents there.

FWIW, I now also tested input streams in the socket client sample (in `DoDownload()`), where `LastRead()` after `Peek()` returns `1`.

But I have also tried `wxZipInputStream` where `LastRead()` after `Peek()` returns `0`.  Not sure why, since `wxZipInputStream` is derived from `wxArchiveInputStream` whose `Peek()` just [returns `wxInputStream::Peek()`](https://github.com/wxWidgets/wxWidgets/blob/cc5f5f90b08b72c7cf21fd4238c6d3bc2e156394/include/wx/archive.h#L102). But maybe I am doing something wrong or `Peek()` does not really make sense for a zipped stream. The test code was basically
```cpp
        wxFileSystem::AddHandler(new wxZipFSHandler);

        const wxString            zipFileName("test.zip#zip:\\test.txt");
        wxFileSystem              zipArchive;
        std::unique_ptr<wxFSFile> zipFile(zipArchive.OpenFile(zipFileName, wxFS_READ));

        if ( zipFile )
        {
            std::unique_ptr<wxInputStream> insz(zipFile->DetachStream());

            if ( insz )
                DoPeek(insm, "wxZipInputStream");
        }
```

**Comment by vadz** [comment]
Thanks for testing, this looks like a bug even though I don't see why does it happen neither from just looking at the code. It would be nice to fix it, but this looks like a separate issue, so I'll create one for it and still merge this one.

**Pull request merged**
